idx/memory/memory.go

```diff
@@ -95,29 +106,32 @@ func ConfigProcess() {
 	}
 }
 
-// interface implemented by both UnpartitionedMemoryIdx and PartitionedMemoryIdx
+// MemoryIndex is an interface implemented by both UnpartitionedMemoryIdx and PartitionedMemoryIdx
 // this is needed to support unit tests.
 type MemoryIndex interface {
 	idx.MetricIndex
-	LoadPartition(int32, []schema.MetricDefinition) int
+	LoadPartition(int32, []idx.MetricDefinition) int
 	UpdateArchive(idx.Archive)
-	add(*schema.MetricDefinition) idx.Archive
+	add(*idx.MetricDefinition) idx.Archive
 	idsByTagQuery(uint32, TagQuery) IdSet
 	PurgeFindCache()
 }
 
+// New returns a Partitioned or Unpartitioned Memory Index
 func New() MemoryIndex {
 	if Partitioned {
 		return NewPartitionedMemoryIdx()
 	}
 	return NewUnpartitionedMemoryIdx()
 }
 
+// Tree is the top level struct in the hierarchical index for a given OrgID
 type Tree struct {
 	Items map[string]*Node // key is the full path of the node.
 }
 
-type IdSet map[schema.MKey]struct{} // set of ids
+// IdSet holds a set of schema.MKeys related to Tag Values
```

Comment:
generally speaking the description of a type should not describe or assume anything about how it will be used.
that we use this type in a way that is related to tag values, doesn't belong in the doc of this type.